pennylane/qinfo/fidelities.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Differentiable quantum fidelity"""
+
+import pennylane as qml
+
+
+def fidelity(qnode0, qnode1, wires0, wires1):
+    r"""Compute the fidelity for two for two :class:`.QNode` returning a :func:`~.state` (a state can be a state vector
+    or a density matrix, depending on the device) acting on quantum systems with the same size. For two pure states, the
+    fidelity corresponds to the squared overlap. For a pure state and a mixed state, it corresponds to the squared
+    expectation of the mixed state in the pure state. Finally for two mixed states, it is defined by the last formula:
+
+    .. math::
```

Comment:
same as in the `math/quantum/` file, I think its better to enumerate equations to reference them or put description - formula alternatingly